### PR TITLE
rados: Make sure Conn isn't GC'd while it has dependant IOContexts.

### PR DESCRIPTION
**Pull request by baergj**
Because we have a finalizer configured for Conn that cleans up Ceph resources, it's important that we don't let it get GC'd before anything that depends on it has been cleaned up. Since clients are free to hold a reference to an IOContext without maintaining a reference to the underlying Conn, we hold such a Conn reference within the IOContext to signal this dependency to the Go GC.

Fixes https://github.com/ceph/go-ceph/issues/569.

### CODE REVIEW

**Comment by phlogistonjohn** [comment]
You are speedy. See my comments on #569 and #662.  @ansiwen and I will need to resume the discussion on this topic once he's available.

That said the patch itself looks OK to me.

**Comment by phlogistonjohn** [comment]
After catching up on the other thread, I am largely in favor of just taking this change. It's simple, it's a very small delta between go-ceph as it is today, and it will fix issues people actually run into.  It may not be the most rigorous option but I think it's the most practical.  However, I won't approve it yet until I see buy in on this PR itself, as I don't want automerge to kick in just yet.

**Review by ansiwen — APPROVED** [review]

**Comment by baergj** [comment]
I've brought the PR up to date with `master` - looks like I'll need re-approvals.

**Review by ansiwen — APPROVED** [review]

**Comment by ansiwen** [comment]
@baergj no need to force-push the rebased versions, else mergify will always remove the approvals. We need to use mergify for the rebase, as long as there are no merge conflicts.

**Review by phlogistonjohn — APPROVED** [review]

**Review by ansiwen — APPROVED** [review]

**Comment by baergj** [comment]
Hmm, I'm not seeing what failed in the Quincy tests. Do either of you know how it failed?

**Comment by baergj** [comment]
Looks like this is ready to merge - do you need anything more from my side?

**Comment by phlogistonjohn** [comment]
@mergifyio rebase

**Comment by phlogistonjohn** [comment]
*sigh* it would help if I remember to put one of the required labels on :-)

**Comment by baergj** [comment]
Thanks all!

**Pull request merged**
